### PR TITLE
RUMM-2533 Internal telemetry

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Implement the several internal telemetry events.

TODO in future PRs : 
 - Automatically send the configuration event
 - add a limitation on telemetry sent to stay within the boundaries of 100 telemetry log / session

### CODE REVIEW

File: library/components/rum/RumAgent.bs [hunk 1]
```diff
@@ -89,17 +89,48 @@ sub addResource(resource as object)
...
+' @param exception (object) the caught exception object
+' ----------------------------------------------------------------
+sub addErrorTelemetry(exception as object)
+    ddLogThread("RumAgent.addErrorTelemetry()")
+    ensureSetup()
+    m.top.telemetryScope@.handleEvent(addTelemetryErrorEvent(exception), m.top.writer)
+end sub
+
+' ----------------------------------------------------------------
+' Adds a telemetry error event
```
Comment (0xnm) [inline library/components/rum/RumAgent.bs:113]:
```suggestion
' Adds a telemetry debug event
```

File: library/components/rum/RumTelemetryScope.bs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+' @param event (object) the event to handle
+' @param writer (object) the writer node (see WriterTask component)
+' ----------------------------------------------------------------
+sub sendDebugEvent(event as object, writer as object)
+    timestamp& = getTimestamp()
+    ddLogVerbose("Sending a debug log telemetry")
+
+    rumContext = m.global.datadogRumContext ?? {}
+
+    configEvent = {
```
Comment (0xnm) [inline library/components/rum/RumTelemetryScope.bs:92]:
```suggestion
    debugEvent = {
```

File: library/components/rum/RumTelemetryScope.bs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+            message: event.message
+        },
+        type: "telemetry",
+        version: sdkVersion(),
+        view: {
+            id: rumContext.viewId
+        }
+    }
+
+    writer.writeEvent = FormatJson(configEvent)
```
Comment (0xnm) [inline library/components/rum/RumTelemetryScope.bs:120]:
```suggestion
    writer.writeEvent = FormatJson(debugEvent)
```

File: library/components/rum/RumTelemetryScope.bs [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+            }
+        },
+        type: "telemetry",
+        version: sdkVersion(),
+        view: {
+            id: rumContext.viewId
+        }
+    }
+
+    writer.writeEvent = FormatJson(configEvent)
```
Comment (0xnm) [inline library/components/rum/RumTelemetryScope.bs:172]:
```suggestion
    writer.writeEvent = FormatJson(errorEvent)
```

File: test/source/tests/rum/Test__RumAgent.bs [hunk 5]
```diff
@@ -169,6 +176,60 @@ function RumAgentTest__WhenAddResource_ThenDelegateToRumScope() as string
...
+
+    ' Then
+    expectedArgs = { event: { eventType: "telemetryConfig", configuration: fakeConfiguration }, writer: m.mockWriter }
+    return m.mockTelemetryScope@.assertFunctionCalled("handleEvent", expectedArgs)
+end function
+
+'----------------------------------------------------------------
+' Given: a RumAgent
+'  When: call the addResource method
+'  Then: delegates to the root RumScope
```
Comment (0xnm) [inline test/source/tests/rum/Test__RumAgent.bs:200]:
```suggestion
'  When: call the addErrorTelemetry method
'  Then: delegates to the root TelemetryScope
```

File: test/source/tests/rum/Test__RumAgent.bs [hunk 6]
```diff
@@ -169,6 +176,60 @@ function RumAgentTest__WhenAddResource_ThenDelegateToRumScope() as string
...
+
+    ' Then
+    expectedArgs = { event: { eventType: "telemetryError", exception: fakeException }, writer: m.mockWriter }
+    return m.mockTelemetryScope@.assertFunctionCalled("handleEvent", expectedArgs)
+end function
+
+'----------------------------------------------------------------
+' Given: a RumAgent
+'  When: call the addResource method
+'  Then: delegates to the root RumScope
```
Comment (0xnm) [inline test/source/tests/rum/Test__RumAgent.bs:218]:
```suggestion
'  When: call the addDebugTelemetry method
'  Then: delegates to the root TelemetryScope
```

**Review by 0xnm — APPROVED** [review]
I left some minor typo/rename suggestions, but overall lgtm!

**Review by mariusc83 — APPROVED** [review]

**Review by kayayarai — APPROVED** [review]

**Pull request merged**